pinot-spi/src/main/java/org/apache/pinot/spi/config/task/AdhocTaskConfig.java

```diff
@@ -57,6 +57,7 @@ public AdhocTaskConfig(@JsonProperty(value = "taskType", required = true) String
       @JsonProperty("taskConfigs") @Nullable Map<String, String> taskConfigs) {
     Preconditions.checkArgument(taskType != null, "'taskType' must be configured");
     Preconditions.checkArgument(tableName != null, "'tableName' must be configured");
+    Preconditions.checkArgument(taskName == null || !taskName.contains("/"), "'taskName' must not contain '/'");
```

Comment:
Can we double check if the task generator explicitly creates the task name or the helix implicitly generates the task name?

If it's the first case, we should add the validation on the task name generated by the task generator as well. If the name gets created by the helix internally, we wouldn't need to worry about this. 

@swaminathanmanish We should consider to remove the task name input from the customer. Ideally, we should internally create the task id.

Reply:
Its created outside Helix by PinotTaskManager.  We can remove the documentation for custom askName and deprecate it?

 if (taskName == null) {
      taskName = tableName + "_" + UUID.randomUUID();
      LOGGER.info("Task name is missing, auto-generate one: {}", taskName);
    }

Reply:
@aishikbh - Could you add this to your message
**taskName' must not contain path separator '/'**

Reply:
> @aishikbh - Could you add this to your message **taskName' must not contain path separator '/'**

Sure, will add that!

Reply:
> @aishikbh Can we add the testing case to cover this? Please check `AdhocTaskConfigTest`

I have added tests related to invalid arguments in AdhocTaskConfig.

> @aishikbh - Could you add this to your message **taskName' must not contain path separator '/'**

I have rephrased the error message as well.